src/connectivity.js

```diff
@@ -0,0 +1,185 @@
+/**
+ * Connectivity Checker to perform basic connectivity checks with BrowserStack
+ * components, i.e. Hub and Rails.
+ * It performs checks with/without proxy, over HTTP(S).
+ * NOTE : HTTPS with Proxy is not implemented since the tool is anyway capturing HTTP
+ * traffic only. Shall be taken up later.
+ */
+
+var http = require('http');
+var url = require('url');
+var constants = require('../config/constants');
+var RdGlobalConfig = constants.RdGlobalConfig;
+var Utils = require('./utils');
+var https = require('https');
+
+/**
+ * Fires the requests to perform connectivity checks
+ * @param {Object} requestOptions 
+ * @param {'http'|'https'} requestType 
+ * @param {String} description 
+ * @param {Array<Number>} successCodes 
+ * @param {Function} callback 
+ */
+var fireRequest = function (requestOptions, requestType, description, successCodes, callback) {
+  var httpOrHttps = requestType === 'http' ? http : https;
```

Comment:
This is an additional overhead to maintain. Why not parse the protocol from the URL itself?

Reply:
We are maintaining only two urls. One for Hub and one for Rails. And that has `http` as prefix. Thus, this is done to generate diff. type of requests while firing, i.e. with/without proxy & with/without HTTPS.

Reply:
Resolved F2F